aggregator_api/src/lib.rs

```diff
@@ -113,13 +117,23 @@ async fn auth_check(conn: &mut Conn, State(cfg): State<Arc<Config>>) -> impl Han
         constant_time::verify_slices_are_equal(bearer_token.as_ref(), key.as_ref()).is_ok()
     }) {
         // Authorization succeeds.
+        conn.set_state(cfg);
```

Comment:
Since it took me a bit to figure out -- I'd appreciate a comment explaining `FromConn for State<T>` takes the config out of the connection state, and this line is putting it back for use by a later handler.

Alternately, we could just access it via `conn.state:<...>()`, and avoid shuffling it around.

Reply:
Tbh I was surprised by this as well. I think it's worth changing how FromConn for State works, and require T be Clone